### ISSUE TITLE
Allow ==- doesn't see everything:

### ISSUE THREAD

**Issue by phadej** [nomeata/inspection-testing#5]
I'm not sure if there's anything one can do (without starting to re implementing simplifier in the library?),
Should we document that `==-` doesn't always work, for example, I get:

```
    LHS:
        ...

        lhsEnum
        lhsEnum
          = \ x ->
              case x of {
                LT -> $WZ `cast` <Co:102>;
                EQ -> lvl `cast` <Co:102>;
                GT -> lvl `cast` <Co:102>
              }
        
        lhsEnum
        lhsEnum = lhsEnum `cast` <Co:303>
        
    RHS:
        ...
        
        rhsEnum
        rhsEnum
          = \ ds ->
              case ds of {
                LT -> $WZ;
                EQ -> lvl;
                GT -> lvl
              }
```

i.e. Quite the same, but there's a coercion in bad place.

*EDIT* the code in question is still under development, but I'll post a link once it's published

**Comment by nomeata** [comment]
I think this is a reasonable request.

We don’t have to reimplement the simplifier, but we'd have to do a better job comparing expression, knowing that the two `lhsEnum`’s are the same (better print this with uniques :-)).

**Comment by phadej** [comment]
The `fin` library in https://github.com/phadej/vec is quite small (and light on dependencies), see https://github.com/phadej/vec/blob/master/fin/test/Inspection.hs

**Comment by nomeata** [comment]
I wanted to reproduce (and fix) it, but If I run the linked file, the error I get for `inspect $  'lhsEnum ==- 'rhsEnum` looks different than the one you posted. I get:
```
    LHS:
        lhsEnum
        lhsEnum = $fEnumOrdering_$sgfrom `cast` ...
        
    RHS:
        lvl_scJI
        lvl_scJI = S @~ ... $WZ
        
        lvl_scJ1
        lvl_scJ1 = S @~ ... lvl_scJI
        
        lvl_scIZ
        lvl_scIZ = S @~ ... $WZ
        
        rhsEnum
        rhsEnum =
          \ ds_dbJk ->
            case ds_dbJk of _ {
              LT -> $WZ;
              EQ -> lvl_scIZ;
              GT -> lvl_scJ1
            }
```
… ah, but only with GHC-8.0. With GHC-8.2 I get the result that you posted.

**Comment by nomeata** [comment]
Fixed in 0.1.2 :-)

**Comment by phadej** [comment]
Looks like GHC-8.0 is quite smart. Cannot blame it for reusing already generated code (from `Data.Fin.Enum`). Subtle behavior, but I guess this library exists to catch such issues :)

Thanks for very prompt fix!
